src/System.Collections/src/System/Collections/Generic/List.cs

```diff
@@ -0,0 +1,1381 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime;
+using System.Runtime.Versioning;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Collections.ObjectModel;
+
+namespace System.Collections.Generic
+{
+    // Implements a variable-size List that uses an array of objects to store the
+    // elements. A List has a capacity, which is the allocated length
+    // of the internal array. As elements are added to a List, the capacity
+    // of the List is automatically increased as required by reallocating the
+    // internal array.
+    // 
+    [DebuggerTypeProxy(typeof(ICollectionDebugView<>))]
+    [DebuggerDisplay("Count = {Count}")]
+    public class List<T> : IList<T>, System.Collections.IList, IReadOnlyList<T>
+    {
+        private const int _defaultCapacity = 4;
+
+        private T[] _items;
+        [ContractPublicPropertyName("Count")]
+        private int _size;
+        private int _version;
+        private Object _syncRoot;
+
+        static readonly T[] _emptyArray = new T[0];
```

Comment:
Style guideline naming: this should be s_emptyArray.  It should also be explicitly marked as private.